Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -224,7 +224,18 @@ private static void HandleSessionError(IntPtr sessionHandlePtr, ErrorCode errorC
                     exception = new SessionException(messageString, errorCode);
                 }
 
-                Session.RaiseError(session, exception);
+                System.Threading.ThreadPool.QueueUserWorkItem(_ =>
+                {
+                    try
+                    {
+                        Session.RaiseError(session, exception);
+                    }
+                    catch { }
+                    finally
+                    {
+                        handle.Dispose();
+                    }
```

Comment:
You should move the outer try/catch here. We don't need the outer one because we're no longer invoking user code, and the one inside the thread pool should log the exception instead of just swallow it.